core/src/HIP/Kokkos_HIP_ReduceScan.hpp

```diff
@@ -49,8 +49,326 @@
 
 #if defined(__HIPCC__)
 
+#include <HIP/Kokkos_HIP_Vectorization.hpp>
```

Comment:
File comment: Can we at least document which parts of this were copied over directly from the Cuda implementation and which parts were changed?

Reply:
I am not sure what you want me to do. We always start from the CUDA code and then we make it work for HIP. We don't come up with new names/new functions. There are probably very few functions that are exactly the same but there are all very close.

Reply:
Adding to that: Things I remember to be different are warp sizes (we are still using 32 explicitly a lot in `CUDA`), the usual renamings (`threadIdx.x` -> `hipThreadIdx.x` and similar), not using deprecated code, simplifications for `KOKKOS_IMPL_CUDA_SHFL_UP_MASK`(and similar), avoiding C-style casts, using `using` instead of `typedef` and possibly everything marked with `FIXME_HIP`. It's really best to just compare the respective files for reviewing even it is tedious.